code/modules/projectiles/boxes_magazines/external_lens.dm

```diff
@@ -0,0 +1,63 @@
+//////////////EXTERNAL LENS//////////////
+
+/obj/item/external_lens
+	name = "external lens"
+	icon_state = "external"
+	desc = "These lens modify the laser frequency and trajectory giving them special effects."
+	icon = 'icons/obj/guns/energy.dmi'
+	var/stored_ammo_type = /obj/item/ammo_casing/energy/laser
+	var/restricted_type = /obj/item/gun/energy/laser //which type of laser gun it can be assigned to
+	var/overlay = "laser"
+
+/obj/item/external_lens/Initialize()
+	. = ..()
+	add_overlay(overlay)
+
+/obj/item/external_lens/afterattack(atom/movable/AM, mob/user, flag)
+	. = ..()
+	if(user && istype(AM, restricted_type) && AM.type == restricted_type)
```

Comment:
the second and the third type are duplicates with the second covering more than the third.